src/resources.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState, useCallback } from 'react';
+import http, { CancelToken } from 'axios';
+import { useAuthentication } from './authentication';
+
+import { fetchUser, updateUser, User } from './api';
+
+export function useUser(id: string) {
+  const [user, setUser] = useState<User | null>(null);
+  const { token } = useAuthentication();
+
+  useEffect(() => {
```

Comment:
Would be clearer to give this function a name.  The purpose of the file is bit hard to interpret, understanding what the function do would help.

Reply:
Can do, haven't seen that used normally (official docs don't do it as well: https://reactjs.org/docs/hooks-effect.html#recap). Here we essentially synchronise the state of our loaded user and the id that we are passed in.  Could call it `synchronizeUser`? Reason it looks scary is mostly because of the crappy `CancelToken` api that's used to cancel API requests, I'm guessing. It's not extremely necessary, but good to have.

Reply:
For me, it's not a hard and fast rule, and not specifically because we're in the context of an effect, although that does add complexity.  Generally, the longer a function gets, the more I think it needs a name.  Because we're in an effect, I have to read and understand what 13 lines of code do.  Once I get to the _end_, I'm like "ok, so all those things I have in my head happen when either id, or token, or setUser change.  It's just a lot to keep in your head.  Even after looking at this twice I don't really understand what it's doing or why.

Mentally it's much easier for me to look at
```
useEffect(doTheThing, [id, token, setUser])
```
and go, "cool, whenever id, token or setUser change, I doTheThing."

I can then think about the implementation of doTheThing independently, but already with some context.  On the other hand, if it was a two line arrow function I wouldn't bother naming it, because it would (probably) be trivial to read.

If it's hard to sum up in one name I would modularise the implementation instead:
```
useEffect(() => {
  doOneThing();
  andTheOtherThing();
}, [id, token, setUser]);
```

Reply:
This would unfortunately make it possibly less readable, because you unfortunately can't build effects like this if you need access to the scope. To make sure that the scope of the effects is correct, you'd first have to do this:
```javascript
useEffect(() => {
  doOneThing();
  andTheOtherThing();
}, [doOneThing, andTheOtherThing]);
```
But as now those functions are dependencies, you need their identity to be guaranteed, so they'd have to be wrapped in `useCallback` with dependencies into the scope, like so:

```javascript
const doOneThing = useCallback(() => {
  // do things with id, token and set user
}, [id, token, setUser]);
```

In addition, that effect does just one thing. All the ceremony around is cleanup and cancellation of the request partway through if it needs to restart.

Reply:
Spent a bit trying to refactor - we decided to focus on this later.